packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -913,15 +913,15 @@ export class Table extends TableBase {
 
     this.encryptionKey = encryptionKey;
 
-    if (props.tableName) { this.node.addMetadata('aws:cdk:hasPhysicalName', props.tableName); }
-
     this.tableArn = this.getResourceArnAttribute(this.table.attrArn, {
       service: 'dynamodb',
       resource: 'table',
       resourceName: this.physicalName,
     });
     this.tableName = this.getResourceNameAttribute(this.table.ref);
 
+    if (props.tableName) { this.node.addMetadata('aws:cdk:hasPhysicalName', this.tableName); }
```

Comment:
For my own understanding, what is this? This is the only place in the CDK this pattern is used

Reply:
I think this was an old idea to mark things with physical names set explicitly in our L2s. I don't think it was adapted widely.

Reply:
>I don't think it was adapted widely.
![Screen Shot 2020-08-03 at 8 02 06 AM](https://user-images.githubusercontent.com/8578043/89196923-c14ed580-d55f-11ea-8615-f9b1c7d6dc06.png)